src/binary/ibc-setup/utils/resolve-option.spec.ts

```diff
@@ -0,0 +1,91 @@
+import test from 'ava';
+
+import { resolveOption } from './resolve-option';
+
+const stringOption1 = 'string option 1';
+const stringOption2 = 'string option 2';
+const functionWithString = () => 'function option';
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const functionWithVoid = () => {};
+
+test('leftmost not undefined option takes precedence', (t) => {
+  const option1 = resolveOption(
+    'first option', // identifier
+    undefined,
+    undefined,
+    stringOption1
+  );
+  t.is(option1, 'string option 1');
```

Comment:
Nice to use the constants in the expectations too.

Reply:
agree 05f0c60